netlogo-gui/src/main/swing/MenuModel.scala

```diff
@@ -29,7 +29,9 @@ class MenuModel[A, B](var groupOrder: Seq[String] = Seq())(implicit leafOrdering
 
   def insertLeaf(leafValue: A, groupName: String = ""): Unit = {
     addGroup(groupName)
-    children = (children :+ Leaf(leafValue, groupName)).sorted
+    val leaf = Leaf(leafValue, groupName)
+    if (!children.contains(leaf))
```

Comment:
But why does it even happen that we try to add a menu item that is already in the list?

Reply:
I changed this code in order to ensure smooth compatibility between the main window menu and the separate code window menu. The way the existing code updated the menus assumed that the only way the menus could change is when a tab is switched, not also when the selected window is changed, so it no longer worked correctly once I added a copy of the menu bar to the separate code window. This duplicate entry check makes the code for synchronizing the two menus much simpler.